code/modules/mob/living/simple_animal/bot/honkbot.dm

```diff
@@ -151,89 +161,83 @@ Maintenance panel panel is [open ? "opened" : "closed"]"},
-		spawn(cooldowntimehorn)
-			spam_flag = 0
+				playsound(src, honksound, 50, 1, -1)
+				spam_flag = TRUE //prevent spam
+		sensor_blink()
+		addtimer(CALLBACK(src, .proc/spam_flag_false), cooldowntimehorn)
 
 	else if (emagged == 2) //emagged honkbots will spam short and memorable sounds.
 
 		if (!client)
-			playsound(loc, pick('sound/items/bikehorn.ogg', 'sound/items/AirHorn2.ogg', 'sound/misc/sadtrombone.ogg', 'sound/items/AirHorn.ogg', 'sound/effects/reee.ogg',  'sound/items/WEEOO1.ogg', 'sound/voice/biamthelaw.ogg', 'sound/voice/bcreep.ogg','sound/magic/Fireball.ogg' ,'sound/effects/pray.ogg', 'sound/voice/hiss1.ogg','sound/machines/buzz-sigh.ogg', 'sound/machines/ping.ogg', 'sound/weapons/flashbang.ogg', 'sound/weapons/bladeslice.ogg'), 50, 0)
-			// to be put at 100 volume: (Weeoo1, bcreep, blaw, hiss, flashbang )
+			playsound(src, "honkbot_e", 50, 0)
 		else
 			if (!spam_flag)
-				playsound(loc, pick('sound/items/bikehorn.ogg', 'sound/items/AirHorn2.ogg', 'sound/misc/sadtrombone.ogg', 'sound/items/AirHorn.ogg', 'sound/effects/reee.ogg', 'sound/items/WEEOO1.ogg', 'sound/voice/biamthelaw.ogg', 'sound/voice/bcreep.ogg','sound/magic/Fireball.ogg' ,'sound/effects/pray.ogg', 'sound/voice/hiss1.ogg','sound/machines/buzz-sigh.ogg', 'sound/machines/ping.ogg', 'sound/weapons/flashbang.ogg', 'sound/weapons/bladeslice.ogg'), 50, 0)
-				spam_flag = 1 // prevent spam
+				playsound(src, "honkbot_e", 50, 0)
+				spam_flag = TRUE // prevent spam
 		icon_state = "honkbot-e"
 		spawn(30) // keep flashing
 			icon_state = "honkbot[on]"
-		spawn(cooldowntimehorn)
-			spam_flag = 0
+		addtimer(CALLBACK(src, .proc/spam_flag_false), cooldowntimehorn)
+
 
 /mob/living/simple_animal/bot/honkbot/proc/honk_attack(mob/living/carbon/C) // horn attack
 	if (!client) //check if a player is controlling
 		playsound(loc, honksound, 50, 1, -1)
 	else
-		if(spam_flag == 0)
+		if(!spam_flag)
 			playsound(loc, honksound, 50, 1, -1)
-			spam_flag = 1 // prevent spam
-	icon_state = "honkbot-c"
-	spawn(5)
-		icon_state = "honkbot[on]"
-	spawn(cooldowntimehorn)
-		spam_flag = 0
+			spam_flag = TRUE // prevent spam
+	sensor_blink()
+	addtimer(CALLBACK(src, .proc/spam_flag_false), cooldowntimehorn)
 
 /mob/living/simple_animal/bot/honkbot/proc/stun_attack(mob/living/carbon/C) // airhorn stun
 	if (!client)
 		playsound(loc, 'sound/items/AirHorn.ogg', 100, 1, -1) //HOOOOOOOOOOOOONK!!
 	else
-		if(spam_flag == 0)
+		if(!spam_flag)
 			playsound(loc, 'sound/items/AirHorn.ogg', 100, 1, -1) //HEEEEEEEEEEEENK!!
-	icon_state = "honkbot-c"
-	spawn(5)
-		icon_state = "honkbot[on]"
+	sensor_blink()
 	if(spam_flag == 0)
 		if(ishuman(C))
 			C.stuttering = 20
+			C.adjustEarDamage(0, 5) //far less damage than the H.O.N.K.
 			C.Jitter(50)
-			C.Knockdown(80)
+			C.Knockdown(60)
 			var/mob/living/carbon/human/H = C
-			if (ckey != null) //don't alter AI behavior
-				spam_flag = 1
+			if(ckey != null) //used over (!ckey) to restore botAI when ghosting.
```

Comment:
I do not think you should ever be checking for a ckey for honkbots, just if there is a client or not. If the player in control of the bot disconnects, the bot will still have a ckey. If you use client checks instead, the bot can resume normal AI behavior until the player reconnects. 

This is, of course, only a recommendation.

Reply:
Using (!client) makes perfect sense, actually.